File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k = k % nums.size(); // Case: k >= nums.size()
        reverse(nums.begin(), nums.end()); // Rotate whole array
        reverse(nums.begin() + k, nums.end()); // Rotate [k, n)
        reverse(nums.begin(), nums.begin() + k); // Rotate [0, k)
    }
};

File: top-interview-150/0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        nums_cpy = nums.copy()
        
        for i in range(0, n):
            pos = (i + k) % n 
            nums[pos] = nums_cpy[i]
        